presentation/matrix_view.py

```diff
@@ -1,153 +1,200 @@
-    if metric == "latency":
-        return config.latency
-    elif metric == "jitter":
-        return config.jitter
-    else:
-        return config.packet_loss
-
-
-def make_data(mesh: MeshResults, metric: str, matrix: Matrix, tresholds: Thresholds) -> List[Dict]:
-    colors = make_colors(mesh, metric, matrix, tresholds)
-    return [
-        dict(
-            x=matrix.agents,
-            y=[i for i in reversed(matrix.agents)],
-            z=colors,
-            text=make_hover_text(mesh, matrix),
-            type="heatmap",
-            hoverinfo="text",
-            opacity=1,
-            name="",
-            showscale=False,
-            colorscale=get_colorscale(colors),
-        )
-    ]
-
-
-def make_colors(mesh: MeshResults, metric: str, matrix: Matrix, tresholds: Thresholds) -> List[List]:
-    colors = []
-    for row in reversed(mesh.rows):
-        colors_col = []
-        for col in row.columns:
-            warning = tresholds.warning(row.agent_id, col.agent_id)
-            error = tresholds.error(row.agent_id, col.agent_id)
-            value = get_metric_value(metric, matrix.cells[row.agent_alias][col.agent_alias])
-            color = get_color(value, warning, error)
-            colors_col.append(color)
-        colors.append(colors_col)
-
-    for i in range(len(mesh.rows)):  # add Nones at diagonal to avoid colorising it
-        colors[-(i + 1)].insert(i, None)  # type: ignore
-
-    return colors
-
-
-def get_metric_value(metric: str, cell: MeshColumn) -> int:
-    if metric == "latency":
-        return cell.latency_microsec.value
-    elif metric == "jitter":
-        return cell.jitter_microsec.value
-    else:
-        return cell.packet_loss_percent.value
-
-
-def make_annotations(mesh: MeshResults, metric: str, matrix: Matrix) -> List[Dict]:
-    annotations = []
-    for row in reversed(mesh.rows):
-        for col in row.columns:
-            text = get_text(metric, matrix.cells[row.agent_alias][col.agent_alias])
-            annotations.append(
-                dict(
-                    showarrow=False,
-                    text=text,
-                    xref="x",
-                    yref="y",
-                    x=col.agent_alias,
-                    y=row.agent_alias,
-                )
-            )
-    return annotations
-
-
-def get_text(metric: str, cell: MeshColumn) -> str:
-    if metric == "latency":
-        return f"<b>{(cell.latency_microsec.value * 1e-3):.2f} ms</b>"
-    elif metric == "jitter":
-        return f"<b>{(cell.jitter_microsec.value * 1e-3):.2f} ms</b>"
-    else:
-        return f"<b>{cell.packet_loss_percent.value:.1f}%</b>"
 
+class MatrixView:
+    MATRIX = "matrix"
+    METRIC_SELECTOR = "metric-selector"
+
+    @staticmethod
+    def make_layout(mesh: MeshResults, metric: Metric) -> html.Div:
+        localtime_timestamp = utc_to_localtime(mesh.utc_timestamp)
+        timestamp = localtime_timestamp.strftime("%H:%M:%S")
```

Comment:
The displayed  timestamp must include date. Imagine if a user loads the page, goes away for  a day, loads the same page in another window/tab after coming back, gets distracted and looks on the one loaded yesterday. She/He has no way to tell which of the two pages is actually current.